discovery-provider/src/tasks/index_rewards_manager_backfill.py

```diff
@@ -581,35 +580,56 @@ def process_solana_rewards_manager(
 
 
 def check_if_backfilling_complete(
-    session: Session, solana_client_manager: SolanaClientManager
+    session: Session, solana_client_manager: SolanaClientManager, redis: Redis
 ) -> bool:
-    stop_sig_tuple = (
-        session.query(IndexingCheckpoint.signature)
-        .filter(
-            IndexingCheckpoint.tablename == index_rewards_manager_backfill_tablename
+    try:
+        redis_complete = redis.get(index_rewards_manager_backfill_complete)
+        if redis_complete:
+            redis_complete = str(redis_complete.decode())
+        if redis_complete == "true":
+            return True
+
+        stop_sig_tuple = (
+            session.query(IndexingCheckpoint.signature)
+            .filter(
+                IndexingCheckpoint.tablename == index_rewards_manager_backfill_tablename
+            )
+            .first()
         )
-        .first()
-    )
-    if not stop_sig_tuple:
+        if not stop_sig_tuple:
+            logger.error(
+                "index_rewards_manager_backfill.py | Tried to check if complete, but no stop_sig"
+            )
+            return False
+        stop_sig = stop_sig_tuple[0]
+
+        one_sig_before_stop_result = solana_client_manager.get_signatures_for_address(
+            REWARDS_MANAGER_PROGRAM,
+            before=stop_sig,
+            limit=1,
+        )
+        if not one_sig_before_stop_result:
+            logger.error("index_rewards_manager_backfill.py | No sigs before stop_sig")
+            return False
+        one_sig_before_stop_result = one_sig_before_stop_result["result"][0]
+        one_sig_before_stop = one_sig_before_stop_result["signature"]
+
+        sig_before_stop_in_db = (
+            session.query(RewardsManagerBackfillTransaction)
+            .filter(RewardsManagerBackfillTransaction.signature == one_sig_before_stop)
+            .first()
+        )
+        complete = bool(sig_before_stop_in_db)
+        redis.set(
+            index_rewards_manager_backfill_complete, "true" if complete else "false"
+        )
```

Comment:
for simplicity, let's just not set the key at all if the backfill isn't complete. also, 1 or 0 probs a bit nicer convention to represent this kind of thing than stringifying 'true'